crates/editor/src/items.rs

```diff
@@ -624,8 +628,18 @@ impl Item for Editor {
             Some(util::truncate_and_trailoff(&description, MAX_TAB_TITLE_LEN))
```

Comment:
It might be better to return the height here instead of updating the `height` variable.

Reply:
Maybe we can also return the `path` so you don't have to determine this twice, see line 632.

Reply:
> Mind elaborating how exactly?

On line 628 we return the description, we can change it to return a tuple of `Some((description, height, path))`. So you can reuse this information on line 639.

Reply:
@RemcoSmitsDev I've tried different things but this seems to be the working way without much complications, returning a tuple made it a bit more complicated and I need to determine the path twice because of the include_filename argument being different. this is what I ended up with, but this is buggy because not all files are showing icons for some reason:
```rs
        let (description, path) = params
            .detail
            .and_then(|detail| {
                let path = path_for_buffer(&self.buffer, detail, false, cx)?;
                let description = path.to_string_lossy();
                let description = description.trim();

                if description.is_empty() {
                    return None;
                }

                let path = path_for_buffer(&self.buffer, detail, true, cx)?;

                Some((
                    util::truncate_and_trailoff(&description, MAX_TAB_TITLE_LEN),
                    path,
                ))
            })
            .unzip();

        let icon = if let Some(path) = path {
            FileIcons::get_icon(&path, cx)
        } else {
            None
        };
```

Reply:
Yeah I'm going off for today, the code in the pr works and isn't that bad imo, I couldn't make any other option (like returning a tuple) work properly (no icons are displayed, no errors), if you have a solution in mind lmk ill take a look at it tmrw.

Reply:
Yeah, I agree, the code is not bad at all. But I thought if we could combine it, so we don't have to determine stuff twice would be cleaner. But how it is fine to.

Reply:
I agree with @RemcoSmitsDev that we could rewrite this in a better way.

I pushed some changes up to the branch that remove the need for the mutable `height` variable.